eureka-client-jersey2/src/main/java/com/netflix/discovery/shared/transport/jersey2/EurekaJersey2ClientImpl.java

```diff
@@ -0,0 +1,354 @@
+import com.netflix.discovery.converters.wrappers.DecoderWrapper;
+import com.netflix.discovery.converters.wrappers.EncoderWrapper;
+import com.netflix.discovery.provider.DiscoveryJerseyProvider;
+import com.netflix.servo.monitor.BasicCounter;
+import com.netflix.servo.monitor.BasicTimer;
+import com.netflix.servo.monitor.Counter;
+import com.netflix.servo.monitor.MonitorConfig;
+import com.netflix.servo.monitor.Monitors;
+import com.netflix.servo.monitor.Stopwatch;
+
+/**
+ * @author Tomasz Bak
+ */
+public class EurekaJersey2ClientImpl implements EurekaJersey2Client {
+
+    private static final Logger s_logger = LoggerFactory.getLogger(EurekaJersey2ClientImpl.class);
+
+    private static final int HTTP_CONNECTION_CLEANER_INTERVAL_MS = 30 * 1000;
+
+    private static final String PROTOCOL = "https";
+    private static final String PROTOCOL_SCHEME = "SSL";
+    private static final int HTTPS_PORT = 443;
+    private static final String KEYSTORE_TYPE = "JKS";
+
+    private final Client apacheHttpClient;
+
+    ClientConfig jerseyClientConfig;
+
+    private final ScheduledExecutorService eurekaConnCleaner =
+            Executors.newSingleThreadScheduledExecutor(new ThreadFactory() {
+
+                private final AtomicInteger threadNumber = new AtomicInteger(1);
+
+                @Override
+                public Thread newThread(Runnable r) {
+                    Thread thread = new Thread(r, "Eureka-Jersey2Client-Conn-Cleaner" + threadNumber.incrementAndGet());
+                    thread.setDaemon(true);
+                    return thread;
+                }
+            });
+
+    public EurekaJersey2ClientImpl(
+            int connectionTimeout,
+            int readTimeout,
+            final int connectionIdleTimeout,
+            ClientConfig clientConfig) {
+
+        try {
+            jerseyClientConfig = clientConfig;
+            jerseyClientConfig.register(DiscoveryJerseyProvider.class);
+            jerseyClientConfig.connectorProvider(new ApacheConnectorProvider());
+            jerseyClientConfig.property(ClientProperties.CONNECT_TIMEOUT, connectionTimeout);
+            jerseyClientConfig.property(ClientProperties.READ_TIMEOUT, readTimeout);
+            apacheHttpClient = ClientBuilder.newClient(jerseyClientConfig);
+            eurekaConnCleaner.scheduleWithFixedDelay(
+                    new ConnectionCleanerTask(connectionIdleTimeout), HTTP_CONNECTION_CLEANER_INTERVAL_MS,
+                    HTTP_CONNECTION_CLEANER_INTERVAL_MS,
+                    TimeUnit.MILLISECONDS);
+        } catch (Throwable e) {
+            throw new RuntimeException("Cannot create Jersey2 client", e);
+        }
+    }
+
+    @Override
+    public Client getClient() {
+        return apacheHttpClient;
+    }
+
+    /**
+     * Clean up resources.
+     */
+    @Override
+    public void destroyResources() {
+        if (eurekaConnCleaner != null) {
+            eurekaConnCleaner.shutdown();
```

Comment:
In the existing code the connCleaner is called one last time before it is shutdown. Do you want to replicate that here too?